net/net/src/TMessage.cxx

```diff
@@ -88,6 +88,11 @@ TMessage::TMessage(void *buf, Int_t bufsize) : TBufferFile(TBuffer::kRead, bufsi
       fBufCompCur = fBuffer + bufsize;
       fBuffer     = nullptr;
       Uncompress();
+      if (adopt) {
+         SetBit(kIsOwnerComp); // bufcomp points to the original buf
+      }
+      // Force being owner of the newly created buffer (inside Uncompress)
+      SetBit(kIsOwner);
```

Comment:
This should be move closer to the allocation (technically that allocation is not quite unconditional and may be reached from other path), so inside `Uncompress`.